pkg/controller/middleware/currentpath.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package middleware defines shared middleware for handlers.
+package middleware
+
+import (
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/google/exposure-notifications-verification-server/pkg/controller"
+	"github.com/gorilla/mux"
+)
+
+type Path struct {
+	uri *url.URL
+}
+
+func CurrentPath() mux.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			ctx := r.Context()
+
+			m := controller.TemplateMapFromContext(ctx)
+			m["currentPath"] = &Path{uri: r.URL}
+
+			ctx = controller.WithTemplateMap(ctx, m)
+			*r = *r.WithContext(ctx)
+
+			next.ServeHTTP(w, r)
+		})
+	}
+}
+
+func (p *Path) Path() string {
```

Comment:
I still think there's going to be cases where we need a strings.Contains on these. What do you think about dangling functions off path like `IsChild(s string)` so we can call those in templates?

Reply:
Well the idea is we have the full path:
/foo/bar/baz -> foo/bar/baz
Dir()
/foo/bar/baz -> /foo/bar
and File()
/foo/bar/baz -> baz

which I think pretty well covers what you'd want to match with eq. You can still do stringContains on any of em

Reply:
Right, I'm more talking about the DevX in the template. Instead of `stringsContains`, we could write these functions to be able to be used like:

```liquid
{{ .currentpath.IsDir("/foo") }}
```

versus

```liquid
{{ stringsContains .currentpath.Dir "/foo" }}
```

Reply:
oh sure we can do that